front/src/components/GoodDeals/index.tsx

```diff
@@ -0,0 +1,118 @@
+import {
+  Box,
+  Button,
+  Paper,
+  Theme,
+  Typography,
+  useMediaQuery,
+} from "@mui/material";
+import { GET_ALL_ARTICLES } from "../../gql/ArticleGql";
+import { useLazyQuery } from "@apollo/client";
+import { ArticleType } from "../../types/article";
+import { useEffect } from "react";
+import Carousel from "react-material-ui-carousel";
+
+export default function GoodDeals() {
+  const isLg = useMediaQuery((theme: Theme) => theme.breakpoints.up("lg"));
+  const isPortrait = useMediaQuery("(orientation: portrait)");
+  const isLandscape = useMediaQuery("(orientation: landscape)");
+
+  const calculateHeight = () => {
+    if (isLg) {
+      return "25vh";
+    } else if (isPortrait) {
+      return "25vh";
+    } else if (isLandscape) {
+      return "35vh";
+    }
```

Comment:
```suggestion
    if (isLg || isPortrait) {
      return "25vh";
    } else if (isLandscape) {
      return "35vh";
    }
```

Même chose, il me semble que calculate height est souvent utilisé en tandem avec les `useMediaQuery`, si on mettait tout ensemble dans un custom hook ça éviterait peut-être d'avoir à recopier ça partout. On peut voir ça ensemble si besoin 👍